docs/source/examples/combine_ddd_zne.md

```diff
@@ -0,0 +1,241 @@
+
+This corresponds to projecting the state onto either the $∣+⟩^{⊗n}$ or $∣−⟩^{⊗n}$ basis. For a perfect GHZ state, this observable will have an expectation value of 1, which corresponds to all qubits being in the same state.
+
+```{code-cell}
+obs = Observable(PauliString("X" * num_qubits))
+print(obs)
+```
+
+For the circuit defined above, the ideal (noiseless) expectation value of the observable is 1:
+
+```{code-cell}
+from functools import partial
+
+ideal_exec = partial(execute, rz_noise = 0.0, depolar_noise = 0.0)
+
+ideal = obs.expectation(circuit, ideal_exec)
+print("Ideal value:", "{:.5f}".format(ideal.real))
+```
+
+The unmitigated (noisy) result however, is impacted by the `Rz` and depolarizing errors. In fact, we are not even halfway to the correct expectation value:
+
+```{code-cell}
+noisy_exec = partial(execute, rz_noise = 0.02, depolar_noise = 0.005)
+noisy = obs.expectation(circuit, noisy_exec) 
+print("Unmitigated noisy value:", "{:.5f}".format(noisy.real))
+```
+
+### Applying Digital Dynamical Decoupling
+
+Next we choose our gate sequences to be used in the digital dynamical decoupling routine (DDD). 
+More information on choosing appropriate sequences can be found in the [DDD theory](../guide/ddd-5-theory.md#common-examples-of-ddd-sequences) section of the user guide.
+
+To do this, we will insert DDD sequences into our circuit itself and then compare the original circuit with the modified one.
+
+```{code-cell}
+from mitiq.ddd import insert_ddd_sequences, rules
+
+print("Original circuit \n", circuit)
+
+rule = rules.yy
+
+ddd_circuit = insert_ddd_sequences(circuit, rule)
+print("DDD modified circuit \n", ddd_circuit)
+```
+
+Now we execute our function on the `ddd_circuit`:
+
+```{code-cell}
+ddd_noisy = obs.expectation(ddd_circuit, noisy_exec)
+
+print("Unmitigated expectation value:", "{:.5f}".format(noisy.real))
+
+print("Expectation value with DDD:", "{:.5f}".format(ddd_noisy.real))
+```
+
+### Zero Noise Extrapolation alone
+
++++
+
+For comparison, we then apply ZNE one our original circuit (without DDD sequences inserted).
+
+```{code-cell}
+from mitiq import zne
+
+zne_executor = zne.mitigate_executor(noisy_exec, observable=obs, scale_noise=zne.scaling.folding.fold_global)
+zne_result = zne_executor(circuit)
+print("Mitigated value obtained with ZNE:", "{:.5f}".format(zne_result.real))
+```
+
+In this case, we see that ZNE by itself actually _makes things worse_ than the unmitigated expectation value!
+
++++
+
+### Digital Dynamical Decoupling + Zero noise extrapolation
+
++++
+
+Finally, we apply a combination of DDD and ZNE.
+DDD is applied first to apply the control pulses to each circuit which ZNE runs to do its extrapolation.
+
+```{code-cell}
+combined_executor = zne.mitigate_executor(execute, observable=obs, scale_noise=zne.scaling.folding.fold_global)
```

Comment:
hmm, shouldn't this use `noisy_exec` instead of `execute`? Otherwise you are comparing scenarios with different levels of noise. And in fact, shouldn't this just be `zne_executor` as defined above. And then you want to apply the ZNE executor to the DDD circuit to combine techniques, i.e.,
```
combined_result = zne_executor(ddd_circuit)
```

Reply:
Thanks, good catch on the `noisy_exec` call, those were comparing two different noise levels. I corrected that, and still find that the performance of DDD + ZNE is better than either individually.

Reply:
True also that I can just call `zne_executor` on the `ddd_circuit` rather than instantiating a new executor called `combined_executor`. They are equivalent as written, but I agree it's unnecessary -- it was a holdover from an earlier way of combining techniques. I agree it's much clearer just to call `zne_executor(ddd_circuit)`. Updated in latest commit.

Reply:
Good to hear that the combined value is still winning!